blog/posts/whats-2024-brought-to-melange-so-far.md

```diff
@@ -0,0 +1,302 @@
+modules are only accessible through the `Js.*` namespace; as a fortunate
+side-effect, we stop exposing `Js__Js_internal`, which could leak into some
+error messages, causing unnecessary confusion.
+
+
+## OCaml 5.2
+
+With OCaml 5.2 around the corner (the first
+[beta](https://github.com/ocaml/ocaml/archive/refs/tags/5.2.0-beta1.tar.gz)
+was released just a couple weeks ago), we made Melange ready for the upcoming
+release. In [melange#1074](https://github.com/melange-re/melange/pull/1074) and
+[melange#1078](https://github.com/melange-re/melange/pull/1078) we upgraded
+both the Melange core and the Stdlib to the changes in OCaml 5.2. As mentioned
+above, we're happy that the next release of Melange will support an even wider
+range of OCaml compiler versions, making 5.2 the latest addition to the
+supported compiler versions.
+
+## Leveraging Dune's potential
+
+### Making Dune faster
+
+Back in January, [Javi](https://twitter.com/javierwchavarri) found a
+performance regression in Dune
+([dune#9738](https://github.com/ocaml/dune/issues/9738)) after upgrading to
+Dune 3.13. The whole fact-finding process of profiling Dune's performance and
+working closely with the team to patch this regression
+([dune#9769](https://github.com/ocaml/dune/pull/9769)) ended up being quite the
+learning experience.
+
+Once the dust settled, Javi took the time to write a [blog
+post](https://tech.ahrefs.com/profiling-dune-builds-a8de589ec268) outlining
+some of the tools he used and the steps he used to gather information about
+Dune's runtime behavior.
+
+
+### Improving `dune describe pp`
+
+The command `dune describe pp` prints a given source file after preprocessing.
+This is useful to quickly inspect the code generate by a (set of) ppx.
+
+`dune describe pp` didn't, however, support [Dune
+dialects](https://dune.readthedocs.io/en/stable/overview.html#term-dialect). I
+[found out](https://github.com/ocaml/dune/issues/4470#issuecomment-1135120774)
+about this limitation when trying to get the preprocessed output of a ReasonML
+file.
+
+We recently set out to fix this problem. In
+[dune#10321](https://github.com/ocaml/dune/pull/10321) we made Reason files and
+dialects generally work within `dune describe pp`, and we followed up with the
+ability to print back the preprocessed output in the same dialect as the given
+file ([dune#10322](https://github.com/ocaml/dune/pull/10322),
+[dune#10339](https://github.com/ocaml/dune/pull/10339) and
+[dune#10340](https://github.com/ocaml/dune/pull/10340)).
+
+
+### [Virtual libraries](https://dune.readthedocs.io/en/stable/variants.html) in Melange
+
+From Dune's own documentation:
+
+> Virtual libraries correspond to Dune’s ability to compile parameterised
+libraries and delay the selection of concrete implementations until linking an
+executable.
+
+In the Melange case there's no executable linking going on, but we can still
+delay the selection of concrete implementations until JS emission – in
```

Comment:
```suggestion
delay the selection of concrete implementations until JavaScript emission – in
```

(there are a few other appearances of JS, I think it's more clear to use the full name so that all audiences are aware of what's being mentioned)